### ISSUE TITLE
Ansible 2.2 no longer decrypts an encrypted inventory

### ISSUE THREAD

**Issue by j16r** [ansible/ansible#18637]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

ansible-vault
ansible-playbook

##### ANSIBLE VERSION

```
ansible 2.2.0.0
  config file =
  configured module search path = Default w/o overrides
```

##### CONFIGURATION

N/A

##### OS / ENVIRONMENT

Darwin junction.local 16.1.0 Darwin Kernel Version 16.1.0: Wed Oct 19 20:31:56 PDT 2016; root:xnu-3789.21.4~4/RELEASE_X86_64 x86_64

##### SUMMARY

In ansible 2.1 we were using an encrypted inventory to do deployments, see the project here: https://github.com/MindLeaps/tracker/tree/master/deploy

In ansible 2.2 with the same command, ansible appears to no longer decrypt the inventory. The playbook doesn't match any hosts. If we change the playbook to match `all` we get an error:

```
fatal: [38303030316563383161373464373362663333376565313664316663336337316131]: UNREACHABLE! => {
    "changed": false,
    "msg": "Failed to connect to the host via ssh: ControlPath too long\r\n",
    "unreachable": true
}
```

It appears to be treating the inventory in its encrypted form as a list of hosts.

##### STEPS TO REPRODUCE

Use an encrypted inventory to deploy.

##### EXPECTED RESULTS

Expect 2.2 to work as 2.1 did. Encrypted inventories should be decrypted when supplied with the correct password.

##### ACTUAL RESULTS
```
No config file found; using defaults
[DEPRECATION WARNING]: Instead of sudo/sudo_user, use become/become_user and make sure become_method is 'sudo' (default).
This feature will be removed in a future release. Deprecation warnings can be disabled by setting
deprecation_warnings=False in ansible.cfg.

PLAYBOOK: playbook.yml *********************************************************
1 plays in playbook.yml

PLAY RECAP *********************************************************************
```

Playbook does not match expected hosts, ignores inventory.

**Comment by alikins** [comment]
This is a duplicate of https://github.com/ansible/ansible/issues/18355

(Fixed in devel in dd0189839eaded82df9e2953600d97272871f2f8 and in stable-2.2 branch in 366bfe14c39d6aee1604928edb2c76589750a516)

**Comment by LearnLinuxTV** [comment]
It looks as though this problem is still happening. I have the same exact problem as in the following report:

https://github.com/ansible/ansible/issues/18355

My inventory file is encrypted, but Ansible cannot parse it despite all variables and the vault-password being installed properly. Worked fine in 2.1, upgrading to 2.2 is the only change. I get errors such as the following:

ansible -m ping all
6165613636856461356662336561633662323866636531356135 | UNREACHABLE! => {
    "changed": false, 
    "msg": "SSH Error: data could not be sent to the remote host. Make sure this host can be reached over ssh", 
    "unreachable": true
}

I am using Ansible 2.2 from jessie-backports in Debian.
https://packages.debian.org/jessie-backports/ansible
